saleor/product/utils.py

```diff
@@ -97,32 +99,61 @@ def products_for_cart(user):
     return products
 
 
-def get_variant_picker_data(variants, attributes):
-    data = {'attributes': [], 'variants': []}
-    for attribute in attributes:
-        data['attributes'].append({
+def get_variant_picker_data(product, discounts=None):
+    availability = get_availability(product, discounts)
+    variants = product.variants.all()
+    data = {'productAttributes': [], 'variantAttributes': [], 'variants': []}
```

Comment:
I think that the real problem here is that we render this UI in React for products with variant attributes or in Django template for products without them. Therefore we need to support two ways of form submission and duplicate parts of the UI.